pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Topic.java

```diff
@@ -137,6 +137,13 @@ default boolean isMarkerMessage() {
                                           long startMessageRollbackDurationSec, boolean replicateSubscriptionState,
                                           KeySharedMeta keySharedMeta);
 
+    /**
+     * Subscribe a topic.
+     * @param option
+     * @return
+     */
+    CompletableFuture<Consumer> subscribe(SubscriptionOption option);
```

Comment:
Looks like we can remove the old one to avoid interface complexity?

Reply:
Deleting the interface is not good, I first set it to @Deprecated